crates/bevy_ecs/src/query/error.rs

```diff
@@ -30,7 +30,22 @@ impl<'w> core::fmt::Display for QueryEntityError<'w> {
                 )?;
                 format_archetype(f, world, entity)
             }
-            Self::NoSuchEntity(entity) => write!(f, "The entity {entity} does not exist"),
+            Self::NoSuchEntity(entity, _world) => {
+                #[cfg(feature = "track_change_detection")]
+                {
+                    if let Some(location) =
+                        _world.entities().get_entity_spawned_despawned_by(entity)
+                    {
+                        write!(f, "The entity {entity} was despawned by {location}",)
+                    } else {
+                        write!(f, "The entity {entity} was never spawned",)
+                    }
+                }
+                #[cfg(not(feature = "track_change_detection"))]
+                {
+                    write!(f, "The entity {entity} does not exist")
```

Comment:
Might be nice to include a hint here to enable `track_change_detection` for a more detailed error message?

Reply:
Yeah that's a good idea. And like the `tracing` filter level features, we should include a note saying library authors should _not_ enable this feature on their published crates, as it removes the ability for end-users to choose themselves.